core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala

```diff
@@ -100,7 +106,13 @@ private[spark] class StreamingPythonRunner(
    */
   def stop(): Unit = {
     pythonWorker.foreach { worker =>
-      SparkEnv.get.destroyPythonWorker(pythonExec, workerModule, envVars.asScala.toMap, worker)
+      val prevConf = conf.get(PYTHON_USE_DAEMON)
+      conf.set(PYTHON_USE_DAEMON, false)
+      try {
+        SparkEnv.get.destroyPythonWorker(pythonExec, workerModule, envVars.asScala.toMap, worker)
+      } finally {
+        conf.set(PYTHON_USE_DAEMON, prevConf)
+      }
```

Comment:
I don't think this change is needed as `env.destroyPythonWorker` won't refer `conf`.